pkg/core/edge_kb.go

```diff
@@ -0,0 +1,296 @@
+package core
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/klothoplatform/klotho/pkg/multierr"
+	"go.uber.org/zap"
+)
+
+type (
+	// Edge defines an entry in a Knowledge base. An Edge represents a valid linking between two types of resources
+	Edge struct {
+		// From represents the source resource in the edge
+		From reflect.Type
```

Comment:
We might want to rename `From` and `To` to `Source` and `Destination` (or `Target`) to be consistent with how we already refer to either end of an edge.